lib/src/domain/communication/api_exception.dart

```diff
@@ -0,0 +1,15 @@
+class ApiException implements Exception {
+  final int code;
+
+  ApiException([this.code]);
+}
+
+class UnauthorizedException extends ApiException {
+  final int code;
+
+  UnauthorizedException(this.code) : super(code);
```

Comment:
w tym momencie nie ma potrzeby trzymania dodatkowo kodu jeżeli jest już w klasie nadrzędnej, ale pytanie czy w tej klasie nadrzędnej powinien być - imo niekoniecznie, skoro tylko 1 implementacja faktycznie ma jakiś statusCode, a co więcej, zastanawiam się czy UnauthorizedException potrzebuje trzymać statusCode, ale to takie do przemyślenia :)

Reply:
no nie musi być status code w tej klasie, to prawda. Imo lepiej, żeby był w klasie nadrzędnej, tych błędów powinno być ogólnie więcej, jakieś 404, 500 wypadałoby dodać. Co do Unauthorized to jest to typowy 401, więc myślę, że jak najbardziej się tu statusCode nadaje